src/zgamma_producer.cpp

```diff
@@ -49,81 +46,154 @@ void ZGammaVarProducer::WriteZGammaVars(nano_tree &nano, pico_tree &pico, vector
+                                pico.out_el_phi()[iel2], 0.000511);
           }
+          lep1.SetPtEtaPhiM(pico.out_el_pt()[iel1], pico.out_el_eta()[iel1], pico.out_el_phi()[iel1], 0.000511);
+          lep2.SetPtEtaPhiM(pico.out_el_pt()[iel2], pico.out_el_eta()[iel2], pico.out_el_phi()[iel2], 0.000511);
+          ptl1err = pico.out_el_energyErr()[iel1] * lep1.Pt() / lep1.P();
+          ptl2err = pico.out_el_energyErr()[iel2] * lep2.Pt() / lep2.P();
+          l1err.SetPtEtaPhiM(pico.out_el_pt()[iel1] + ptl1err,
+                             pico.out_el_eta()[iel1], pico.out_el_phi()[iel1], 0.000511);
+          l2err.SetPtEtaPhiM(pico.out_el_pt()[iel2] + ptl2err,
+                             pico.out_el_eta()[iel2], pico.out_el_phi()[iel2], 0.000511);
+	  
+          dml1 = (l1err + lep2 + photon).M() - (lep1 + lep2 + photon).M();
+          dml2 = (lep1 + l2err + photon).M() - (lep1 + lep2 + photon).M();
+          dmph = (lep1 + lep2 + pherr).M() - (lep1 + lep2 + photon).M();
         }
         else {
           int imu1 = pico.out_ll_i1()[ill];
           int imu2 = pico.out_ll_i2()[ill];
           if(pico.out_mu_charge()[imu1] < 0) {
             lminus.SetPtEtaPhiM(pico.out_mu_pt()[imu1], pico.out_mu_eta()[imu1],
-                                pico.out_mu_phi()[imu1],0.105);
+                                pico.out_mu_phi()[imu1], 0.10566);
             lplus.SetPtEtaPhiM(pico.out_mu_pt()[imu2], pico.out_mu_eta()[imu2],
-                               pico.out_mu_phi()[imu2],0.105);
+                               pico.out_mu_phi()[imu2], 0.10566);
           }
           else {
             lplus.SetPtEtaPhiM(pico.out_mu_pt()[imu1], pico.out_mu_eta()[imu1],
-                               pico.out_mu_phi()[imu1],0.105);
+                               pico.out_mu_phi()[imu1], 0.10566);
             lminus.SetPtEtaPhiM(pico.out_mu_pt()[imu2], pico.out_mu_eta()[imu2],
-                                pico.out_mu_phi()[imu2],0.105);
+                                pico.out_mu_phi()[imu2], 0.10566);
           }
+          lep1.SetPtEtaPhiM(pico.out_mu_pt()[imu1], pico.out_mu_eta()[imu1], pico.out_mu_phi()[imu1], 0.10566);
+          lep2.SetPtEtaPhiM(pico.out_mu_pt()[imu2], pico.out_mu_eta()[imu2], pico.out_mu_phi()[imu2], 0.10566);
+          ptl1err = pico.out_mu_ptErr()[imu1];
+          ptl2err = pico.out_mu_ptErr()[imu2];
+          l1err.SetPtEtaPhiM(pico.out_mu_pt()[imu1] + ptl1err,
+                             pico.out_mu_eta()[imu1], pico.out_mu_phi()[imu1], 0.10566);
+          l2err.SetPtEtaPhiM(pico.out_mu_pt()[imu2] + ptl2err,
+                             pico.out_mu_eta()[imu2], pico.out_mu_phi()[imu2], 0.10566);
+          dml1 = (l1err + lep2 + photon).M() - (lep1 + lep2 + photon).M();
+          dml2 = (lep1 + l2err + photon).M() - (lep1 + lep2 + photon).M();
+          dmph = (lep1 + lep2 + pherr).M() - (lep1 + lep2 + photon).M();
         }
+        pico.out_llphoton_dml1().push_back(dml1);
```

Comment:
Could we have better names for dml1, dml2 and dmph?

Reply:
I could change "llphoton_dml1" to "llphoton_l1_masserr". Any suggestions?